sycl/doc/extensions/supported/sycl_ext_intel_fpga_device_selector.asciidoc

```diff
@@ -0,0 +1,120 @@
+= sycl_ext_fpga_selector
+
+:source-highlighter: coderay
+:coderay-linenums-mode: table
+
+// This section needs to be after the document title.
+:doctype: book
+:toc2:
+:toc: left
+:encoding: utf-8
+:lang: en
+:dpcpp: pass:[DPC++]
+
+// Set the default source code type in this document to C++,
+// for syntax highlighting purposes.  This is needed because
+// docbook uses c++ and html5 uses cpp.
+:language: {basebackend@docbook:c++:cpp}
+
+
+== Notice
+
+[%hardbreaks]
+Copyright (C) 2022-2022 Intel Corporation.  All rights reserved.
+
+Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are trademarks
+of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc. used by
+permission by Khronos.
+
+
+== Contact
+
+To report problems with this extension, please open a new issue at:
+
+https://github.com/intel/llvm/issues
+
+
+== Dependencies
+
+This extension is written against the SYCL 2020 revision 5 specification.  All
+references below to the "core SYCL specification" or to section numbers in the
+SYCL specification refer to that revision.
+
+== Status
+
+This extension is implemented and fully supported by {dpcpp}.
+
+[NOTE]
+====
+This extension is currently implemented in {dpcpp} only for FPGA devices.
+In the {dpcpp} implementation, the application must be built with
+"-Xshardware" in order to use an FPGA hardware device.
+In the {dpcpp} implementation, the application must be built with
+"-Xssimulation" in order to use an FPGA simulator device.
+====
+
+
+== Overview
+
+This extension provides selectors to choose between FPGA hardware devices, FPGA
+simulator devices, and FPGA emulator devices. The current implementation is
+based on the platform name.  This is useful in the most common case when the
+user has one FPGA board installed in their system (one device per platform).
+
+To use these selectors, add
+....
+#include <sycl/ext/intel/fpga_device_selector.hpp>
+....
+to your source code.
+
+
+== Specification
+
+=== Select FPGA hardware device
+....
+// force FPGA hardware device
+sycl::queue deviceQueue{sycl::ext::intel::fpga_selector{}};
+....
+
+=== Select FPGA simulator device
+....
+// force FPGA simulator device
+sycl::queue deviceQueue{sycl::ext::intel::fpga_simulator_selector{}};
+....
+
```

Comment:
I'd add a sentence here like:

> Added in version 2 of this extension.

I know the feature-test table says this also, but readers are much more likely to notice the sentence here.

Reply:
done